internal/scheduleringester/instructions.go

```diff
@@ -314,9 +318,18 @@ func (c *InstructionConverter) handleCancelJob(cancelJob *armadaevents.CancelJob
 	}}, nil
 }
 
-func (c *InstructionConverter) handleCancelJobSet(jobset string) ([]DbOperation, error) {
+func (c *InstructionConverter) handleCancelJobSet(cancelJobSet *armadaevents.CancelJobSet, meta eventSequenceCommon) ([]DbOperation, error) {
+	cancelQueued := len(cancelJobSet.States) == 0 || slices.Contains(cancelJobSet.States, armadaevents.JobState_QUEUED)
+	cancelLeased := len(cancelJobSet.States) == 0 || slices.Contains(cancelJobSet.States, armadaevents.JobState_PENDING) || slices.Contains(cancelJobSet.States, armadaevents.JobState_RUNNING)
```

Comment:
shouldn't it be AND rather than OR here a otherwise sending a an instruction to cancel the pending will cancel the running too?  Or is this just for compatibility with the old system?

Reply:
I've done it this was for compatibility - but I agree AND would be more sensible

However it does leave you in an awkward place if you request just leased and then we do nothing.

Probably the API is wrong and should just be swapped to only allowing cancelling queued/leased.